### ISSUE TITLE
Error Building Rock from source

### ISSUE THREAD

**Issue by oraqlle** [Champii/Rock#80]
I was trying to download Rock using the cargo+git command and during compilation I got two errors. I've included the output below.

```zsh
error[E0658]: deriving `Default` on enums is experimental
 --> src/lib/diagnostics/diagnostics_list.rs:9:17
  |
9 | #[derive(Debug, Default, Clone)]
  |                 ^^^^^^^
  |
  = note: see issue #86985 <https://github.com/rust-lang/rust/issues/86985> for more information
  = note: this error originates in the derive macro `Default` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0554]: `#![feature]` may not be used on the stable release channel
 --> src/lib/rock.rs:1:1
  |
1 | #![feature(associated_type_bounds)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Some errors have detailed explanations: E0554, E0658.
For more information about an error, try `rustc --explain E0554`.
error: could not compile `rock` due to 2 previous errors
error: failed to compile `rock v0.2.2 (https://github.com/Champii/Rock#97e477ae)`, intermediate artifacts can be found at `/tmp/cargo-installmX4vWY`
```

It seems you've used an experimental feature that not all people with a rust/cargo installation might have, for lack of a better phrase access to because they are not using the nightly builds of rust. Just thought I'd point that out.

**Comment by Champii** [comment]
You might be right, indeed !
I'm going to check if this feature is really needed. If not, I'll remove it.
Thank you

**Comment by Champii** [comment]
Fixed in 4598b28b4cd553e179a75d95b9f9fbf089f4eb2e
